work_queue/src/work_queue.c

```diff
@@ -697,6 +700,47 @@ work_queue_msg_code_t process_info(struct work_queue *q, struct work_queue_worke
 	return MSG_PROCESSED;
 }
 
+/*
+A cache-update message coming from the worker means that a requested
+remote transfer or command was successful, and know we know the size
+of the file for the purposes of cache storage management.
+*/
+
+int process_cache_update( struct work_queue *q, struct work_queue_worker *w, const char *line )
+{
+	char cachename[WORK_QUEUE_LINE_MAX];
+	long long size;
+	long long transfer_time;
+	
+	if(sscanf(line,"cache-update %s %lld %lld",cachename,&size,&transfer_time)==3) {
+		struct stat *remote_info = hash_table_lookup(w->current_files,cachename);
+		if(remote_info) {
+			remote_info->st_size = size;
+			remote_info->st_mtime = time(0);
+			// XXX store the transfer time once we have a better structure here
+		}
+	}
+	
+	return MSG_PROCESSED;
+}
+
+/*
+A cache-invalid message coming from the worker means that a requested
+remote transfer or command did not succeed, and the intended file is
+not in the cache.  So, we remove the corresponding note for that worker.
+We should expect to soon receive some failed tasks that were unable
+set up their own input sandboxes.
+*/
+
+int process_cache_invalid( struct work_queue *q, struct work_queue_worker *w, const char *line )
+{
+	char cachename[WORK_QUEUE_LINE_MAX];
+	if(sscanf(line,"cache-invalid %s",cachename)==1) {
```

Comment:
I wonder if we should send the stdout/stderr of the commands that failed to retrieve the file. E.g.:
cache-invalid %s %d, with the stdout+stderr follows, of that %d size.  Otherwise it may be hard to debug.

Reply:
Yes, I like that idea.

Reply:
Done.